src/Interpreters/Context.h

```diff
@@ -424,6 +425,9 @@ class ContextData
     /// mutation tasks of one mutation executed against different parts of the same table.
     PreparedSetsCachePtr prepared_sets_cache;
 
+    /// Cache for storages resolved during query processing.
```

Comment:
Ok, but also add a clarification - why do we have to resolve each table only once.